man-embedder.c

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+/* Copyright (c) 2024 Elastic NV */
+
+#include <err.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#include <unistd.h>
+
+static void
+usage(void)
+{
+	fprintf(stderr, "usage: %s input_file ifdef_name\n",
+	    program_invocation_short_name);
+
+	exit(1);
+}
+
+static int
+embed(const char *input_path, const char *ifdef_name)
+{
+	FILE	*input;
+	int	 ch, line_wrap;
+
+	if ((input = fopen(input_path, "r")) == NULL)
+		err(1, "fopen");
+
+	if (ifdef_name != NULL)
```

Comment:
Is this ever true? If not, perhaps we should bail here instead?

Reply:
I wanted it to be a bit more flexible, but yeah, not really used, I was thinking I'd end up copying this code to other places in the future where an ifdef would not be needed.